src/coreclr/jit/codegenriscv64.cpp

```diff
@@ -7210,9 +7186,24 @@ inline void CodeGen::genJumpToThrowHlpBlk_la(
 
         if (addr == nullptr)
         {
-            NYI_RISCV64("part of genJumpToThrowHlpBlk_la-----unimplemented on RISCV64 yet----");
             callType   = emitter::EC_INDIR_R;
             callTarget = REG_DEFAULT_HELPER_CALL_TARGET;
+            if (compiler->opts.compReloc)
+            {
+                ssize_t imm = (3 + 1) << 2;
+                emit->emitIns_R_R_I(ins, EA_PTRSIZE, reg1, reg2, imm);
+                emit->emitIns_R_AI(INS_jal, EA_PTR_DSP_RELOC, callTarget, (ssize_t)pAddr);
+            }
+            else
+            {
+                ssize_t imm = (9 + 7) << 2;
+                emit->emitIns_R_R_I(ins, EA_PTRSIZE, reg1, reg2, imm);
+                // TODO: In the future we may consider using emitter::emitLoadImmediate instead,
+                // which is less straightforward but offers slightly better codegen.
+                emitLoadImmediateDirectly(GetEmitter(), callTarget, (ssize_t)pAddr);
+                emit->emitIns_R_R_I(INS_ld, EA_PTRSIZE, callTarget, callTarget, 0);
```

Comment:
emitLoadImmediateDirectly already derefs at the end, you need one more?

Reply:
Good catch, I think it's wrong, but let me check emitted assembly first.